lightning/src/routing/scoring.rs

```diff
@@ -1702,6 +1547,166 @@ mod approx {
 	}
 }
 
+mod bucketed_history {
```

Comment:
Is having these in a submodule that useful? Note in another comment that I left, it may actually be better to get rid of `HistoricalMinMaxBuckets` in order to share `now` and `last_updated`.

Reply:
Hmm... I think it would make more sense separate out all the liquidity code and keep the historical tracking with it.

Reply:
> Aside from the core concept of "how do I convert liquidity bounds into a probability" (which I plan to move into a util method in a coming PR), they don't really share anything - they're doing very different calculation.

That's the penalty computation. But tracking and updating the current liquidity seems to have a relationship with historical liquidity in so far they are both code related to liquidity, even if the code itself doesn't overlap. I mean, structurally they belong together at very least given one is a sub-struct of the other. Seems odd to have those in separate modules.

> Personally, I find `scorer.rs` pretty hard to follow, there are too many structs flying around tracking liquidity bounds, current liquidity bounds, etc. If we can split it into separate, isolated (which requires a module in rust, module members can always read struct fields otherwise) bits it'd read much cleaner. I don't feel super strongly about a new file, but a module at least makes sense, and probably putting the log table in a new file cause its really distracting.

FWIW, when I said "liquidity code and keep the historical tracking with it", I meant move `ChannelLiquidity`, `DirectedChannelLiquidity`, and all the historical tracking structs into a separate module.

Reply:
> That's a large part of the file, though? Aside from the traits themselves and the log tables and tests of the same.

Yeah, implementation code is about 600 lines excluding log approximation (about 800 with). Only a few of the tests examine the liquidity structs directly. The rest look at `channel_penalty_msat`.

I'm fine putting that in one file / module and having `scoring.rs` use whatever interface that it exposes.